src/reducers/gotPlayer/histogram.js

```diff
@@ -0,0 +1,33 @@
+import { playerHistogramActions } from 'actions';
+import createReducer from 'reducers/reducerFactory';
+
+const initialHistogramState = {
+  list: [],
+  error: false,
+  loading: true,
+  loaded: false,
+};
+
+const initialState = {
+  [undefined]: initialHistogramState,
+};
+
+export default createReducer(initialState, playerHistogramActions, true);
+
+export const getPlayerHistogram = {
+  getPlayerHistogramById: (state, id) => {
+    if (!state.app.gotPlayer.histogram.byId[id]) {
+      return {
+        ...initialState,
+      };
+    }
+    return state.app.gotPlayer.histogram.byId[id];
+  },
+  getHistogram: histogramName => (state, id) => getPlayerHistogram.getPlayerHistogramById(state, id)[histogramName] || initialHistogramState,
```

Comment:
I think there will only ever be one histogram in state at any given time.  You can just make a request on each click and swap the histograms in state.